### PR TITLE
Add author data to the docs collection

### PR DESCRIPTION
**Pull request by sulenchy**
## Description

Add author data to the docs collection

### CODE REVIEW

**Review by jaaneh — CHANGES_REQUESTED** [review]
@sulenchy can you restore the package-lock file?

Also, can we handle displaying the Author component conditionally if there's an author included? Some case studies don't include one and will simply display _`Author: `_ without a name.

The actual h1 title loses some of it's styling also, wonder if we can restore the styling it has currently

File: src/components/CustomPageTitle.astro [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+const { title, author, draft } = Astro.props.entry.data;
+---
+
+<header>
+  <h1>{title}</h1>
+  <em>Author: {author}</em>
```
Comment (jaaneh) [inline src/components/CustomPageTitle.astro:7]:
```suggestion
  {author && (
    <em>Author: {author}</em>
  )}
```

**Review by jaaneh — CHANGES_REQUESTED** [review]

File: src/components/CustomPageTitle.astro [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+</header>
```
Comment (jaaneh) [inline src/components/CustomPageTitle.astro:8]:
```suggestion
---
const { title, author } = Astro.props.entry.data;
---

<h1 id="_top">{title}</h1>
{author && (
  <em>Author: {author}</em>
)}

<style>
  h1 {
    margin-top: 1rem;
    font-size: var(--sl-text-h1);
    line-height: var(--sl-line-height-headings);
    font-weight: 600;
    color: var(--sl-color-white);
  }
</style>
```

Let's conditionally render the author. This also restores default styling to the h1, both by applying manually and removing the wrapping <header>

**Review by jaaneh — CHANGES_REQUESTED** [review]

**Comment by sulenchy** [comment]
> @sulenchy can you restore the package-lock file?

Thanks for this comment @jaaneh. Did not notice it. I am restoring it ASAP.

**Pull request merged**